src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -1675,8 +1676,19 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       //
       //   // Later generated code expects foo.bar = {} to exist:
       //   foo.bar.Baz = function() { /* ... */ }
-      printer->Print("goog.exportSymbol('$name$', null, global);\n", "name",
-                     *it);
+
+      // Do not use global scope in strict mode
+      if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
+        string namespaceObject = *it;
+        // Remove "proto." from the namespace object
+        namespaceObject.erase(0, 6);
+        GOOGLE_CHECK(namespaceObject.find('proto.'));
```

Comment:
@haberman now?

Reply:
I'm confused, it looks like you are doing the check after the removal? If we are asserting that we are removing "proto.", shouldn't we check *before* the removal?

How is the JavaScript test passing in Travis? Is this code-path being tested?

Reply:
I moved the check before removal, please check it out.

Reply:
In the most recent version of the code, I still see:

```
+        namespaceObject.erase(0, 6);
+        GOOGLE_CHECK(namespaceObject.find('proto.'));
```

Am I missing something?

I am still worried that we are not exercising this code-path, since the JavaScript test is passing despite this code being incorrect.